ios/NativeBridge.swift

```diff
@@ -0,0 +1,181 @@
+        }
+        if let value = organization {
+            let _ = builder.organization(value)
+        }
+        if let value = invitationUrl, let invitationURL = URL(string: value) {
+            let _ = builder.invitationURL(invitationURL)
+        }
+        if(leeway != 0) {
+            let _ = builder.leeway(leeway)
+        }
+        if(ephemeralSession) {
+            let _ = builder.useEphemeralSession()
+        }
+        builder.parameters(additionalParameters)
+        builder.start { result in
+            switch result {
+            case .success(let credentials):
+                resolve(credentials.asDictionary())
+            case .failure(let error):
+                reject(NativeBridge.errorCode, error.errorDescription, error)
+            }
+        }
+            
+    }
+    
+    @objc public func webAuthLogout(scheme: String?, federated: Bool, resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
+        let builder = Auth0.webAuth(clientId: self.clientId, domain: self.domain)
+        builder.clearSession(federated: federated) { result in
+                switch result {
+                case .success:
+                    resolve(true)
+                case .failure(let error):
+                    reject(NativeBridge.errorCode, error.errorDescription, error)
+                }
+            }
+    }
+    
+    @objc public func saveCredentials(credentialsDict: [String: Any], resolve: RCTPromiseResolveBlock, reject: RCTPromiseRejectBlock) {
+        
+        guard let accessToken = credentialsDict[NativeBridge.accessTokenKey] as? String, let tokenType = credentialsDict[NativeBridge.tokenTypeKey] as? String, let idToken = credentialsDict[NativeBridge.idTokenKey] as? String else { reject(NativeBridge.errorCode, "Incomplete information provided for credentials", NSError.init(domain: NativeBridge.errorCode, code: -99999, userInfo: nil)); return; }
+        
+        let refreshToken = credentialsDict[NativeBridge.refreshTokenKey] as? String
+        let scope = credentialsDict[NativeBridge.scopeKey] as? String
+        var expiresIn: Date?
+         if let string = credentialsDict[NativeBridge.expiresInKey] as? String, let double = Double(string) {
+             expiresIn = Date(timeIntervalSinceNow: double)
+         } else if let double = credentialsDict[NativeBridge.expiresInKey] as? Double {
+             expiresIn = Date(timeIntervalSinceNow: double)
+         } else if let dateStr = credentialsDict[NativeBridge.expiresInKey] as? String {
+             let dateFormatter = DateFormatter()
+             dateFormatter.dateFormat = NativeBridge.dateFormat
+             expiresIn = dateFormatter.date(from: dateStr)
+         }
+     
+        
+        let credentials =  Credentials(
+            accessToken: accessToken,
+            tokenType: tokenType,
+            idToken:  idToken,
+            refreshToken: refreshToken,
+            expiresIn: expiresIn ?? Date(),
```

Comment:
If `expiresIn` is `nil`, should it be an error instead of using the current date?